### PR TITLE
Use pytest-randomly for reproducible random test parameters

### PR DESCRIPTION
**Pull request by pavoljuhas**
- Require pytest-randomly for the testing to ensure RNG-generated test
  parameters are consistent across parallel test jobs.
- Remove check/pytest option --actually-quiet which disables pytest-randomly
  hook for seeding parallel test jobs.
- Remove `CIRQ_TESTING_RANDOM_SEED` as it is not needed anymore.
- Remove RNG seeding in the tests where it seems redundant.
- Clean up one instance of unnecessary test parametrization.

This upholds #4787 and replaces #4788.
Reverts #1825 and obsoletes #1826.

### CODE REVIEW

**Review by vtomole — APPROVED** [review]
Thanks for the ping. Forgot to submit review: LGTM.

**Comment by vtomole** [comment]
What happened to https://github.com/quantumlib/Cirq/pull/4788#issuecomment-1005220812?

**Comment by vtomole** [comment]
https://github.com/quantumlib/Cirq/issues/5880

**Comment by pavoljuhas** [comment]
> What happened to [#4788 (comment)](https://github.com/quantumlib/Cirq/pull/4788#issuecomment-1005220812)?

I believe I was checking the tests with check/pytest script which was executing `pytest --quiet ...`.
pytest-randomly needs to do seeding _before_ the test discovery so that the seed is active at import time for the RNG-dependent test parameters.  This happens in the [pytest_report_header hook](https://github.com/pytest-dev/pytest-randomly/blob/243932e61d4a0411a6360f30f87d348678bdf622/src/pytest_randomly/__init__.py#L188), which is however disabled by the `--quiet` option.

In short, `pytest --quiet` breaks seeding of test parameters by pytest-randomly and with that also parallel testing (test jobs get different tests).  Things are fine without the `-q, --quiet` option, so this PR removes it in 13890c62815188116fa57b9432cd99fd5036e175.
